api/src/main/java/io/grpc/NameResolverRegistry.java

```diff
@@ -47,9 +47,10 @@
 
   @GuardedBy("this")
   private final LinkedHashSet<NameResolverProvider> allProviders = new LinkedHashSet<>();
-  /** Immutable, sorted version of {@code allProviders}. Is replaced instead of mutating. */
```

Comment:
Why delete the "is replaced instead of mutating" comment? That seems all the more important now that it is a LinkedHashMap and so can't easily be made unmodifiable.